### ISSUE TITLE
Curriculum/lesson page is broken

### ISSUE THREAD

**Issue by Ulisseus** [garageScript/c0d3-app#669]
Going to `https://www.c0d3.com/curriculum/2` results in error screen. Something changed `Session` type so it can return null under submission field hence `_.get(session, 'submissions', [])` does not work anymore.

**Comment by Ulisseus** [comment]
TLDR: Our typescript types don't work because of manual type assertions `X as Y` and `X:Y`. 

We switched from incorrect manual types to almost useless codegen ones, case in point `pages/curriculum/[lesson].tsx` file:
```js
 const userSubmissions: Submission[] = (_.get(session, 'submissions', []) as Submission[])
  ```
 This breaks production but it passes typescript checks because we manually added type instead of letting typescript to infer it. I'm not really know where to start with fixing it, initial typedefs are still not strict enough.
